File: stat_tool.py
from pathlib import Path
import datetime
from check_file_permission import check_file_permission

def stat_file(file_path, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.exists():
        raise ValueError(f"File not found: {file_path}")
    stat = file_path.stat()
    return (f"File: {file_path}\nSize: {stat.st_size} bytes\n"\
            f"Modified: {datetime.datetime.fromtimestamp(stat.st_mtime).isoformat()}")

File: ts-tools/stat_tool.ts
import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function statFile(filePath: string, config: any): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'read', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.stat(resolvedPath, (err, stats) => {
          if (err) {
            reject(err);
          } else {
            resolve(
              `File: ${resolvedPath}\nSize: ${stats.size} bytes\nModified: ${stats.mtime.toISOString()}`,
            );
          }
        });
      })
      .catch(reject);
  });
}
